In [1]:
import pandas as pd
import numpy as np


Here Reading the dataframe

In [2]:
df = pd.read_csv("C:\\Users\\ASUS\\Downloads\\data sets\\ensemble\\creditcard.csv")

In [3]:
df

Time         V1         V2        V3        V4        V5  \
0            0.0  -1.359807  -0.072781  2.536347  1.378155 -0.338321   
1            0.0   1.191857   0.266151  0.166480  0.448154  0.060018   
2            1.0  -1.358354  -1.340163  1.773209  0.379780 -0.503198   
3            1.0  -0.966272  -0.185226  1.792993 -0.863291 -0.010309   
4            2.0  -1.158233   0.877737  1.548718  0.403034 -0.407193   
...          ...        ...        ...       ...       ...       ...   
284802  172786.0 -11.881118  10.071785 -9.834783 -2.066656 -5.364473   
284803  172787.0  -0.732789  -0.055080  2.035030 -0.738589  0.868229   
284804  172788.0   1.919565  -0.301254 -3.249640 -0.557828  2.630515   
284805  172788.0  -0.240440   0.530483  0.702510  0.689799 -0.377961   
284806  172792.0  -0.533413  -0.189733  0.703337 -0.506271 -0.012546   

              V6        V7        V8        V9  ...       V21       V22  \
0       0.462388  0.239599  0.098698  0.363787  ... -0.018307  0.277838   
1      -0.082361 -0.078803  0.085102 -0.255425  ... -0.225775 -0.638672   
2       1.800499  0.791461  0.247676 -1.514654  ...  0.247998  0.771679   
3       1.247203  0.237609  0.377436 -1.387024  ... -0.108300  0.005274   
4       0.095921  0.592941 -0.270533  0.817739  ... -0.009431  0.798278   
...          ...       ...       ...       ...  ...       ...       ...   
284802 -2.606837 -4.918215  7.305334  1.914428  ...  0.213454  0.111864   
284803  1.058415  0.024330  0.294869  0.584800  ...  0.214205  0.924384   
284804  3.031260 -0.296827  0.708417  0.432454  ...  0.232045  0.578229   
284805  0.623708 -0.686180  0.679145  0.392087  ...  0.265245  0.800049   
284806 -0.649617  1.577006 -0.414650  0.486180  ...  0.261057  0.643078   

             V23       V24       V25       V26       V27       V28  Amount  \
0      -0.110474  0.066928  0.128539 -0.189115  0.133558 -0.021053  149.62   
1       0.101288 -0.339846  0.167170  0.125895 -0.008983  0.014724    2.69   
2       0.909412 -0.689281 -0.327642 -0.139097 -0.055353 -0.059752  378.66   
3      -0.190321 -1.175575  0.647376 -0.221929  0.062723  0.061458  123.50   
4      -0.137458  0.141267 -0.206010  0.502292  0.219422  0.215153   69.99   
...          ...       ...       ...       ...       ...       ...     ...   
284802  1.014480 -0.509348  1.436807  0.250034  0.943651  0.823731    0.77   
284803  0.012463 -1.016226 -0.606624 -0.395255  0.068472 -0.053527   24.79   
284804 -0.037501  0.640134  0.265745 -0.087371  0.004455 -0.026561   67.88   
284805 -0.163298  0.123205 -0.569159  0.546668  0.108821  0.104533   10.00   
284806  0.376777  0.008797 -0.473649 -0.818267 -0.002415  0.013649  217.00   

        Class  
0           0  
1           0  
2           0  
3           0  
4           0  
...       ...  
284802      0  
284803      0  
284804      0  
284805      0  
284806      0  

[284807 rows x 31 columns]

Let's check the NaN/ Null/ Missing in the dataset

In [4]:
df.isna().sum()

Time      0
V1        0
V2        0
V3        0
V4        0
V5        0
V6        0
V7        0
V8        0
V9        0
V10       0
V11       0
V12       0
V13       0
V14       0
V15       0
V16       0
V17       0
V18       0
V19       0
V20       0
V21       0
V22       0
V23       0
V24       0
V25       0
V26       0
V27       0
V28       0
Amount    0
Class     0
dtype: int64

Let's check zero's in the dependent columns only 

In [5]:
def checkzero(df):
    return df[df==0].value_counts()

In [6]:
df.columns

Index(['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10',
       'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20',
       'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount',
       'Class'],
      dtype='object')

In [7]:
clist = ['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10',
       'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20',
       'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount']

In [8]:
for i in clist:
    print(checkzero(df[i]))

0.0    2
Name: Time, dtype: int64
Series([], Name: V1, dtype: int64)
Series([], Name: V2, dtype: int64)
Series([], Name: V3, dtype: int64)
Series([], Name: V4, dtype: int64)
Series([], Name: V5, dtype: int64)
Series([], Name: V6, dtype: int64)
Series([], Name: V7, dtype: int64)
Series([], Name: V8, dtype: int64)
Series([], Name: V9, dtype: int64)
Series([], Name: V10, dtype: int64)
Series([], Name: V11, dtype: int64)
Series([], Name: V12, dtype: int64)
Series([], Name: V13, dtype: int64)
Series([], Name: V14, dtype: int64)
Series([], Name: V15, dtype: int64)
Series([], Name: V16, dtype: int64)
Series([], Name: V17, dtype: int64)
Series([], Name: V18, dtype: int64)
Series([], Name: V19, dtype: int64)
Series([], Name: V20, dtype: int64)
Series([], Name: V21, dtype: int64)
Series([], Name: V22, dtype: int64)
Series([], Name: V23, dtype: int64)
Series([], Name: V24, dtype: int64)
Series([], Name: V25, dtype: int64)
Series([], Name: V26, dtype: int64)
Series([], Name: V27, dtype: int64)
Ser

Here am not going to fill the zeros of Amount feature b'c amount may zero and zero present in time feature going to drop it  

In [9]:
df["Time"][df["Time"]==0].index

Int64Index([0, 1], dtype='int64')

In [10]:
dff = df.drop([0, 1])

In [11]:
dff

Time         V1         V2        V3        V4        V5  \
2            1.0  -1.358354  -1.340163  1.773209  0.379780 -0.503198   
3            1.0  -0.966272  -0.185226  1.792993 -0.863291 -0.010309   
4            2.0  -1.158233   0.877737  1.548718  0.403034 -0.407193   
5            2.0  -0.425966   0.960523  1.141109 -0.168252  0.420987   
6            4.0   1.229658   0.141004  0.045371  1.202613  0.191881   
...          ...        ...        ...       ...       ...       ...   
284802  172786.0 -11.881118  10.071785 -9.834783 -2.066656 -5.364473   
284803  172787.0  -0.732789  -0.055080  2.035030 -0.738589  0.868229   
284804  172788.0   1.919565  -0.301254 -3.249640 -0.557828  2.630515   
284805  172788.0  -0.240440   0.530483  0.702510  0.689799 -0.377961   
284806  172792.0  -0.533413  -0.189733  0.703337 -0.506271 -0.012546   

              V6        V7        V8        V9  ...       V21       V22  \
2       1.800499  0.791461  0.247676 -1.514654  ...  0.247998  0.771679   
3       1.247203  0.237609  0.377436 -1.387024  ... -0.108300  0.005274   
4       0.095921  0.592941 -0.270533  0.817739  ... -0.009431  0.798278   
5      -0.029728  0.476201  0.260314 -0.568671  ... -0.208254 -0.559825   
6       0.272708 -0.005159  0.081213  0.464960  ... -0.167716 -0.270710   
...          ...       ...       ...       ...  ...       ...       ...   
284802 -2.606837 -4.918215  7.305334  1.914428  ...  0.213454  0.111864   
284803  1.058415  0.024330  0.294869  0.584800  ...  0.214205  0.924384   
284804  3.031260 -0.296827  0.708417  0.432454  ...  0.232045  0.578229   
284805  0.623708 -0.686180  0.679145  0.392087  ...  0.265245  0.800049   
284806 -0.649617  1.577006 -0.414650  0.486180  ...  0.261057  0.643078   

             V23       V24       V25       V26       V27       V28  Amount  \
2       0.909412 -0.689281 -0.327642 -0.139097 -0.055353 -0.059752  378.66   
3      -0.190321 -1.175575  0.647376 -0.221929  0.062723  0.061458  123.50   
4      -0.137458  0.141267 -0.206010  0.502292  0.219422  0.215153   69.99   
5      -0.026398 -0.371427 -0.232794  0.105915  0.253844  0.081080    3.67   
6      -0.154104 -0.780055  0.750137 -0.257237  0.034507  0.005168    4.99   
...          ...       ...       ...       ...       ...       ...     ...   
284802  1.014480 -0.509348  1.436807  0.250034  0.943651  0.823731    0.77   
284803  0.012463 -1.016226 -0.606624 -0.395255  0.068472 -0.053527   24.79   
284804 -0.037501  0.640134  0.265745 -0.087371  0.004455 -0.026561   67.88   
284805 -0.163298  0.123205 -0.569159  0.546668  0.108821  0.104533   10.00   
284806  0.376777  0.008797 -0.473649 -0.818267 -0.002415  0.013649  217.00   

        Class  
2           0  
3           0  
4           0  
5           0  
6           0  
...       ...  
284802      0  
284803      0  
284804      0  
284805      0  
284806      0  

[284805 rows x 31 columns]

let's check outliers on the basis of the skewness of the column.

In [12]:
def outliersiqr(d,j):
    q1 = d[j].quantile(0.25)
    q3 = d[j].quantile(0.75)
    iqr = q3 - q1
    low = q1 - (1.5*iqr)
    hi = q3 + (1.5*iqr)
    l1 = d[(d[j]<= low) | (d[j]>= hi)]
    v1 = d[(d[j]>= low) & (d[j]<= hi)]
    return l1,v1

In [13]:
def outliesmsd(d,j):
    m = d[j].mean()
    sd = d[j].std()
    low = m - (3*sd)
    hi = m + (3*sd)
    ot= d[(d[j]<=low) | (d[j]>=hi)]
    vld =  d[(d[j]>=low) & (d[j]<=hi)] 
    return ot, vld

In [14]:
for i in clist:
    outl,valid = outliersiqr(dff,i)

In [15]:
outl

Time        V1        V2        V3        V4        V5        V6  \
2            1.0 -1.358354 -1.340163  1.773209  0.379780 -0.503198  1.800499   
20          16.0  0.694885 -1.361819  1.029221  0.834159 -1.191209  1.309109   
51          36.0 -1.004929 -0.985978 -0.038039  3.710061 -6.631951  5.122103   
64          42.0 -0.522666  1.009923  0.276470  1.475289 -0.707013  0.355243   
85          55.0 -4.575093 -4.429184  3.402585  0.903915  3.002224 -0.491078   
...          ...       ...       ...       ...       ...       ...       ...   
284735  172727.0 -1.661169 -0.565425  0.294268 -1.549156 -2.301359  2.365956   
284748  172738.0  1.634178 -0.486939 -1.975967  0.495364  0.263635 -0.713049   
284753  172743.0  1.465737 -0.618047 -2.851391  1.425282  0.893893 -0.958325   
284757  172745.0 -1.757643 -0.982659  1.091540 -1.409539 -0.662159  0.046930   
284806  172792.0 -0.533413 -0.189733  0.703337 -0.506271 -0.012546 -0.649617   

              V7        V8        V9  ...       V21       V22       V23  \
2       0.791461  0.247676 -1.514654  ...  0.247998  0.771679  0.909412   
20     -0.878586  0.445290 -0.446196  ... -0.295583 -0.571955 -0.050881   
51      4.371691 -2.006868 -0.278736  ...  1.393406 -0.381671  0.969719   
64      1.559849 -0.399579 -0.479813  ...  0.172401  1.011543  0.069666   
85     -2.705393  0.666451  1.922216  ... -0.047365  0.853360 -0.971600   
...          ...       ...       ...  ...       ...       ...       ...   
284735 -0.248881 -0.857361  0.137784  ...  1.432397 -0.257828 -0.072471   
284748  0.459925 -0.336879  0.743676  ... -0.113197 -0.493594  0.001993   
284753  1.508074 -0.625691 -0.369824  ...  0.500426  0.870491 -0.495410   
284757  0.173241  0.364812  1.143381  ... -0.191471 -0.432979 -0.292549   
284806  1.577006 -0.414650  0.486180  ...  0.261057  0.643078  0.376777   

             V24       V25       V26       V27       V28   Amount  Class  
2      -0.689281 -0.327642 -0.139097 -0.055353 -0.059752   378.66      0  
20     -0.304215  0.072001 -0.422234  0.086553  0.063499   231.71      0  
51      0.019445  0.570923  0.333278  0.857373 -0.075538  1402.95      0  
64      0.157820 -1.109224 -0.302369  0.318170  0.316910   243.66      0  
85     -0.114862  0.408300 -0.304576  0.547785 -0.456297   200.01      0  
...          ...       ...       ...       ...       ...      ...    ...  
284735 -1.035804 -0.437889 -0.238543  0.365302 -0.448621   381.05      0  
284748  0.602533 -0.049936 -0.145522 -0.040554  0.024884   220.28      0  
284753  0.676929  0.787688 -0.254918 -0.117200 -0.014737   337.54      0  
284757 -0.281494 -0.484349  0.366751 -0.102628 -0.458915   200.00      0  
284806  0.008797 -0.473649 -0.818267 -0.002415  0.013649   217.00      0  

[31915 rows x 31 columns]

In [16]:
valid

Time         V1         V2        V3        V4        V5  \
3            1.0  -0.966272  -0.185226  1.792993 -0.863291 -0.010309   
4            2.0  -1.158233   0.877737  1.548718  0.403034 -0.407193   
5            2.0  -0.425966   0.960523  1.141109 -0.168252  0.420987   
6            4.0   1.229658   0.141004  0.045371  1.202613  0.191881   
7            7.0  -0.644269   1.417964  1.074380 -0.492199  0.948934   
...          ...        ...        ...       ...       ...       ...   
284801  172785.0   0.120316   0.931005 -0.546012 -0.745097  1.130314   
284802  172786.0 -11.881118  10.071785 -9.834783 -2.066656 -5.364473   
284803  172787.0  -0.732789  -0.055080  2.035030 -0.738589  0.868229   
284804  172788.0   1.919565  -0.301254 -3.249640 -0.557828  2.630515   
284805  172788.0  -0.240440   0.530483  0.702510  0.689799 -0.377961   

              V6        V7        V8        V9  ...       V21       V22  \
3       1.247203  0.237609  0.377436 -1.387024  ... -0.108300  0.005274   
4       0.095921  0.592941 -0.270533  0.817739  ... -0.009431  0.798278   
5      -0.029728  0.476201  0.260314 -0.568671  ... -0.208254 -0.559825   
6       0.272708 -0.005159  0.081213  0.464960  ... -0.167716 -0.270710   
7       0.428118  1.120631 -3.807864  0.615375  ...  1.943465 -1.015455   
...          ...       ...       ...       ...  ...       ...       ...   
284801 -0.235973  0.812722  0.115093 -0.204064  ... -0.314205 -0.808520   
284802 -2.606837 -4.918215  7.305334  1.914428  ...  0.213454  0.111864   
284803  1.058415  0.024330  0.294869  0.584800  ...  0.214205  0.924384   
284804  3.031260 -0.296827  0.708417  0.432454  ...  0.232045  0.578229   
284805  0.623708 -0.686180  0.679145  0.392087  ...  0.265245  0.800049   

             V23       V24       V25       V26       V27       V28  Amount  \
3      -0.190321 -1.175575  0.647376 -0.221929  0.062723  0.061458  123.50   
4      -0.137458  0.141267 -0.206010  0.502292  0.219422  0.215153   69.99   
5      -0.026398 -0.371427 -0.232794  0.105915  0.253844  0.081080    3.67   
6      -0.154104 -0.780055  0.750137 -0.257237  0.034507  0.005168    4.99   
7       0.057504 -0.649709 -0.415267 -0.051634 -1.206921 -1.085339   40.80   
...          ...       ...       ...       ...       ...       ...     ...   
284801  0.050343  0.102800 -0.435870  0.124079  0.217940  0.068803    2.69   
284802  1.014480 -0.509348  1.436807  0.250034  0.943651  0.823731    0.77   
284803  0.012463 -1.016226 -0.606624 -0.395255  0.068472 -0.053527   24.79   
284804 -0.037501  0.640134  0.265745 -0.087371  0.004455 -0.026561   67.88   
284805 -0.163298  0.123205 -0.569159  0.546668  0.108821  0.104533   10.00   

        Class  
3           0  
4           0  
5           0  
6           0  
7           0  
...       ...  
284801      0  
284802      0  
284803      0  
284804      0  
284805      0  

[252901 rows x 31 columns]

In [17]:
for i in clist:
    o,vv =outliesmsd(dff,i)

In [18]:
o

Time        V1         V2         V3        V4         V5  \
51          36.0 -1.004929  -0.985978  -0.038039  3.710061  -6.631951   
89          59.0 -0.773293  -4.146007  -0.932038  0.027094  -1.698307   
140         87.0 -5.101877   1.897022  -3.458034 -1.277543  -5.517758   
150         93.0 -1.640014  -0.479831   1.613630  2.997979  -3.803788   
164        103.0 -6.093248 -12.114213  -5.694973  3.294389  -1.413792   
...          ...       ...        ...        ...       ...        ...   
284249  172273.0 -9.030538 -11.112584 -16.233798  3.592021 -40.427726   
284290  172305.0  0.311762  -2.602832  -2.942923  1.880832  -0.090804   
284383  172397.0 -0.065120  -4.114668  -2.106854 -0.137339  -1.384530   
284497  172498.0 -0.801222  -4.874934  -3.195666  1.177224  -1.299393   
284528  172527.0  0.643223  -3.380753  -3.890511 -0.386292  -0.266371   

               V6         V7        V8        V9  ...       V21       V22  \
51       5.122103   4.371691 -2.006868 -0.278736  ...  1.393406 -0.381671   
89       0.460188   0.737344 -0.314216 -0.842673  ...  0.890978  0.026123   
140      2.098366   3.329603  1.250966  0.271501  ... -0.871744 -0.678879   
150      5.051812   3.315254 -0.303785  0.671395  ... -0.367933  0.641891   
164      4.776000   4.808426 -0.228197 -0.525896  ...  2.228823 -2.264037   
...           ...        ...       ...       ...  ...       ...       ...   
284249  23.917837  44.054461 -7.277778 -4.210637  ... -0.269048  0.988144   
284290  -0.480139   1.865720 -0.549267 -0.303862  ...  0.718151  0.220383   
284383   0.239999   0.893833 -0.257922  1.145888  ...  0.395868 -1.244969   
284497  -0.397073   2.192428 -0.603173  0.358854  ...  0.778143 -1.011155   
284528  -0.822619   1.514705 -0.729189 -0.901784  ...  0.944202  0.691763   

             V23       V24       V25       V26       V27       V28    Amount  \
51      0.969719  0.019445  0.570923  0.333278  0.857373 -0.075538   1402.95   
89     -1.134769 -0.654958  0.098386 -0.209150 -0.171709  0.208057   1142.02   
140    -0.555900 -0.761660  0.066611  0.767227  0.731634 -0.860310    919.60   
150    -0.405779 -1.268700  0.494090  0.610967  0.608993 -0.410034    937.69   
164    -3.553381  1.215279 -0.406073 -0.653384 -0.711222  0.672900   3828.04   
...          ...       ...       ...       ...       ...       ...       ...   
284249  7.040028  0.347693  2.520869  2.342495  3.478175 -2.713136  10199.44   
284290 -0.913711 -0.332605  0.278026 -0.474820 -0.200782  0.073386    897.00   
284383 -0.686403 -0.963878 -0.924685  0.210599 -0.253408  0.125687   1114.00   
284497 -1.160277 -0.266300 -0.523929 -0.233779 -0.338745  0.180252   1484.66   
284528 -1.051281  0.140447  0.508555  0.220468 -0.278604  0.058006    900.00   

        Class  
51          0  
89          0  
140         0  
150         0  
164         0  
...       ...  
284249      0  
284290      0  
284383      0  
284497      0  
284528      0  

[4076 rows x 31 columns]

In [19]:
vv

Time         V1         V2        V3        V4        V5  \
2            1.0  -1.358354  -1.340163  1.773209  0.379780 -0.503198   
3            1.0  -0.966272  -0.185226  1.792993 -0.863291 -0.010309   
4            2.0  -1.158233   0.877737  1.548718  0.403034 -0.407193   
5            2.0  -0.425966   0.960523  1.141109 -0.168252  0.420987   
6            4.0   1.229658   0.141004  0.045371  1.202613  0.191881   
...          ...        ...        ...       ...       ...       ...   
284802  172786.0 -11.881118  10.071785 -9.834783 -2.066656 -5.364473   
284803  172787.0  -0.732789  -0.055080  2.035030 -0.738589  0.868229   
284804  172788.0   1.919565  -0.301254 -3.249640 -0.557828  2.630515   
284805  172788.0  -0.240440   0.530483  0.702510  0.689799 -0.377961   
284806  172792.0  -0.533413  -0.189733  0.703337 -0.506271 -0.012546   

              V6        V7        V8        V9  ...       V21       V22  \
2       1.800499  0.791461  0.247676 -1.514654  ...  0.247998  0.771679   
3       1.247203  0.237609  0.377436 -1.387024  ... -0.108300  0.005274   
4       0.095921  0.592941 -0.270533  0.817739  ... -0.009431  0.798278   
5      -0.029728  0.476201  0.260314 -0.568671  ... -0.208254 -0.559825   
6       0.272708 -0.005159  0.081213  0.464960  ... -0.167716 -0.270710   
...          ...       ...       ...       ...  ...       ...       ...   
284802 -2.606837 -4.918215  7.305334  1.914428  ...  0.213454  0.111864   
284803  1.058415  0.024330  0.294869  0.584800  ...  0.214205  0.924384   
284804  3.031260 -0.296827  0.708417  0.432454  ...  0.232045  0.578229   
284805  0.623708 -0.686180  0.679145  0.392087  ...  0.265245  0.800049   
284806 -0.649617  1.577006 -0.414650  0.486180  ...  0.261057  0.643078   

             V23       V24       V25       V26       V27       V28  Amount  \
2       0.909412 -0.689281 -0.327642 -0.139097 -0.055353 -0.059752  378.66   
3      -0.190321 -1.175575  0.647376 -0.221929  0.062723  0.061458  123.50   
4      -0.137458  0.141267 -0.206010  0.502292  0.219422  0.215153   69.99   
5      -0.026398 -0.371427 -0.232794  0.105915  0.253844  0.081080    3.67   
6      -0.154104 -0.780055  0.750137 -0.257237  0.034507  0.005168    4.99   
...          ...       ...       ...       ...       ...       ...     ...   
284802  1.014480 -0.509348  1.436807  0.250034  0.943651  0.823731    0.77   
284803  0.012463 -1.016226 -0.606624 -0.395255  0.068472 -0.053527   24.79   
284804 -0.037501  0.640134  0.265745 -0.087371  0.004455 -0.026561   67.88   
284805 -0.163298  0.123205 -0.569159  0.546668  0.108821  0.104533   10.00   
284806  0.376777  0.008797 -0.473649 -0.818267 -0.002415  0.013649  217.00   

        Class  
2           0  
3           0  
4           0  
5           0  
6           0  
...       ...  
284802      0  
284803      0  
284804      0  
284805      0  
284806      0  

[280729 rows x 31 columns]

Let's check the skewness of every column

In [20]:
def skewness(df):
    return df.skew()

In [21]:
for i in clist:
    print(skewness(dff[i]))

-0.03555515604203246
-3.2806710287115632
-4.6248487958229285
-2.2401783937640882
0.6763036927012193
-2.425895842615213
1.8265783059331275
2.5539003540273546
-8.521913189830801
0.5546792581882649
1.1871358859310728
0.35650749489592387
-2.278409868876342
0.06523139020112735
-1.9951750457982305
-0.3084183175334819
-1.100965762589501
-3.844901445477721
-0.25988136599380174
0.10919446886896324
-2.037147001611598
3.5929769790216657
-0.2132609498049071
-5.8751216447915695
-0.5525024440774826
-0.4157856817750667
0.5766900348505846
-1.1702023876807042
11.192052051167467
16.97767975851137


functions are made to remove the outliers

In [22]:
def capiqr(d,j,v):
    q1 = d[j].quantile(0.25)
    q3 = d[j].quantile(0.75)
    iqr = q3 - q1
    low = q1 - (1.5*iqr)
    hi = q3 + (1.5*iqr)
    vl =  d[(d[j]>=low) & (d[j]<=hi)]
    m = vl[j].mean()
    d[j]=d[j].apply(lambda x : m if x<=low  else m if x>=hi else x )


In [23]:
def capmsd(d,j):
    m = d[j].mean()
    sd = d[j].std()
    low = m - (3*sd)
    hi = m + (3*sd)
    ot= d[(d[j]<=low) | (d[j]>=hi)]
    vld =  d[(d[j]>=low) & (d[j]<=hi)]
    m = vld[i].mean()
    d[j]=d[j].apply(lambda x : m if x<=low  else m if x>=hi else x )
    

Calling the function with respect to skeawness of the columns

In [24]:
for i in clist:
    if (dff[i].skew()<0.8) or (dff[i].skew()>-0.8):
        capmsd(dff,i)
        
    else:
        cap(dff,i)
        
        

After removing the outliers once again checking the skwewness 

In [25]:
def skewness(df):
    return df.skew()

In [26]:
for i in clist:
    print(skewness(dff[i]))

-0.03555515604203246
-0.6763107837783908
-0.23688056012696138
-0.36638104953574907
0.2644184974454129
0.36145359959170237
1.3195301712596614
-0.05672118600584595
-0.5570660565548321
0.10481184715405509
0.2763808472337311
0.07882726794416801
-0.8431328722838438
0.03533040161701818
-0.16754743041853468
-0.25181656833926375
-0.2837840364530668
0.5697550169024633
0.02213309355118483
-0.07912013175244725
0.8996595105295306
0.282449172125511
0.0008198028363646447
0.3881546870015325
-0.5895508739463471
-0.13247558491837694
0.3855069589149657
-0.31556170820872287
-0.8464269753693741
3.2626468699334894


In [27]:
dff.sample(284805)

Time        V1        V2        V3        V4        V5        V6  \
49640    44145.0 -0.696396 -0.241229 -0.408676 -1.740249  1.457204  3.393790   
222848  143142.0  1.627987 -1.205290  0.107450  0.842445 -1.371615  0.030433   
192949  129926.0  2.193130 -1.095908 -1.171685 -1.182291 -0.480923  0.122549   
199952  133211.0  1.863803 -0.654008 -0.412362  0.093165 -0.653742 -0.137965   
110266   71748.0  1.293096  0.228450 -0.420786  0.608169  0.625865  0.354944   
...          ...       ...       ...       ...       ...       ...       ...   
214400  139660.0  2.254795 -0.916685 -1.747627 -1.335973 -0.032580  0.012549   
256684  157808.0  2.102655 -0.974302 -1.179432 -0.807065 -0.840413 -1.049734   
18992    29935.0  0.644142 -0.631311  0.109628  1.608971 -0.386484 -0.075987   
114411   73471.0  1.090836  0.049330  0.389106  1.235535 -0.003552  0.578478   
232025  147015.0  2.096896 -1.047092 -0.498208 -0.985978 -0.990114 -0.246691   

              V7        V8        V9  ...       V21       V22       V23  \
49640  -0.804324  1.292347 -1.823703  ...  0.086013 -0.390562  0.238211   
222848 -0.873133  0.155639  2.221969  ...  0.048792  0.165293  0.088432   
192949 -0.995175  0.070281 -0.024317  ...  0.178963  0.483516 -0.005648   
199952 -0.609632  0.029443  1.148072  ...  0.254206  0.869341  0.024408   
110266  0.108094  0.013017 -0.180206  ... -0.023482 -0.076611 -0.340328   
...          ...       ...       ...  ...       ...       ...       ...   
214400 -0.737479  0.049812 -0.385963  ...  0.313038  0.771852 -0.008176   
256684 -0.489680 -0.131114 -0.316009  ...  0.020935 -0.061061  0.226102   
18992   0.446043 -0.101943  0.214222  ...  0.064958 -0.174398 -0.303147   
114411 -0.199736  0.336911  0.201085  ... -0.039109  0.084770 -0.023899   
232025 -1.056771  0.059539 -0.129948  ... -0.002589 -0.065904  0.317133   

             V24       V25       V26       V27       V28  Amount  Class  
49640   0.979593 -0.372631 -0.399055  0.013027  0.034578   88.80      0  
222848 -0.110614 -0.351419 -0.373286  0.043571 -0.002747  165.61      0  
192949 -1.383863 -0.053275 -0.111235 -0.012739 -0.072577   28.80      0  
199952 -0.266204 -0.149939  0.116657  0.006760 -0.043565   64.95      0  
110266 -1.382353  0.886826 -0.240799  0.005565 -0.004344   19.95      0  
...          ...       ...       ...       ...       ...     ...    ...  
214400 -0.364303  0.117562 -0.015295 -0.048121 -0.079432    8.40      0  
256684  0.031298 -0.191185 -0.349560 -0.052855 -0.068943   40.00      0  
18992   0.116821  0.574264 -0.342558 -0.009125  0.063621  278.00      0  
114411 -0.310561  0.521931 -0.295466  0.046776 -0.001379    1.00      0  
232025 -0.496352 -0.502531 -0.476254  0.007034 -0.051710   29.97      0  

[284805 rows x 31 columns]

Now lets split in X and y

In [28]:
X = dff.drop("Class", axis = 1)

In [29]:
X

Time        V1        V2        V3        V4        V5        V6  \
2            1.0 -1.358354 -1.340163  1.773209  0.379780 -0.503198  1.800499   
3            1.0 -0.966272 -0.185226  1.792993 -0.863291 -0.010309  1.247203   
4            2.0 -1.158233  0.877737  1.548718  0.403034 -0.407193  0.095921   
5            2.0 -0.425966  0.960523  1.141109 -0.168252  0.420987 -0.029728   
6            4.0  1.229658  0.141004  0.045371  1.202613  0.191881  0.272708   
...          ...       ...       ...       ...       ...       ...       ...   
284802  172786.0  0.124553  0.056222  0.052136 -2.066656  0.007328 -2.606837   
284803  172787.0 -0.732789 -0.055080  2.035030 -0.738589  0.868229  1.058415   
284804  172788.0  1.919565 -0.301254 -3.249640 -0.557828  2.630515  3.031260   
284805  172788.0 -0.240440  0.530483  0.702510  0.689799 -0.377961  0.623708   
284806  172792.0 -0.533413 -0.189733  0.703337 -0.506271 -0.012546 -0.649617   

              V7        V8        V9  ...       V20       V21       V22  \
2       0.791461  0.247676 -1.514654  ...  0.524980  0.247998  0.771679   
3       0.237609  0.377436 -1.387024  ... -0.208038 -0.108300  0.005274   
4       0.592941 -0.270533  0.817739  ...  0.408542 -0.009431  0.798278   
5       0.476201  0.260314 -0.568671  ...  0.084968 -0.208254 -0.559825   
6      -0.005159  0.081213  0.464960  ... -0.219633 -0.167716 -0.270710   
...          ...       ...       ...  ...       ...       ...       ...   
284802  0.012668  0.065668  1.914428  ...  1.475829  0.213454  0.111864   
284803  0.024330  0.294869  0.584800  ...  0.059616  0.214205  0.924384   
284804 -0.296827  0.708417  0.432454  ...  0.001396  0.232045  0.578229   
284805 -0.686180  0.679145  0.392087  ...  0.127434  0.265245  0.800049   
284806  1.577006 -0.414650  0.486180  ...  0.382948  0.261057  0.643078   

             V23       V24       V25       V26       V27       V28  Amount  
2       0.909412 -0.689281 -0.327642 -0.139097 -0.055353 -0.059752  378.66  
3      -0.190321 -1.175575  0.647376 -0.221929  0.062723  0.061458  123.50  
4      -0.137458  0.141267 -0.206010  0.502292  0.219422  0.215153   69.99  
5      -0.026398 -0.371427 -0.232794  0.105915  0.253844  0.081080    3.67  
6      -0.154104 -0.780055  0.750137 -0.257237  0.034507  0.005168    4.99  
...          ...       ...       ...       ...       ...       ...     ...  
284802  1.014480 -0.509348  1.436807  0.250034  0.943651  0.823731    0.77  
284803  0.012463 -1.016226 -0.606624 -0.395255  0.068472 -0.053527   24.79  
284804 -0.037501  0.640134  0.265745 -0.087371  0.004455 -0.026561   67.88  
284805 -0.163298  0.123205 -0.569159  0.546668  0.108821  0.104533   10.00  
284806  0.376777  0.008797 -0.473649 -0.818267 -0.002415  0.013649  217.00  

[284805 rows x 30 columns]

In [30]:
y = dff["Class"]

In [31]:
y

2         0
3         0
4         0
5         0
6         0
         ..
284802    0
284803    0
284804    0
284805    0
284806    0
Name: Class, Length: 284805, dtype: int64

In [32]:
from sklearn.model_selection import train_test_split

In [33]:
Xtrain,Xtest,ytrain,ytest = train_test_split(X,y,test_size= 0.70)

In [34]:
Xtrain.shape

(85441, 30)

In [35]:
Xtest.shape

(199364, 30)

In [36]:
ytrain.shape

(85441,)

In [37]:
ytest.shape

(199364,)

In [38]:
from sklearn.ensemble import AdaBoostClassifier

In [39]:
abc = AdaBoostClassifier()

In [40]:
abc.fit(Xtrain,ytrain)

AdaBoostClassifier()

In [41]:
abc.score(Xtest,ytest)

0.9986155976003692

In [42]:
abc.score(Xtrain,ytrain)

0.998923233576386

Here our dataset is completely is imbalanced b'c equal no of clussified target

In [43]:
dff["Class"].value_counts()

0    284313
1       492
Name: Class, dtype: int64

Here we have 284313 zeros and only 492 ones but for making a good model all the classified data (""Target) should be nearly equal in counting. So in order to make a good model we a balanced data and to make the data balnced let's take no of zero's equal to no of one's 

In [44]:
Z = dff[dff["Class"]==0].sample(492)

In [45]:
Z

Time        V1        V2        V3        V4        V5        V6  \
225550  144293.0  2.161229  0.206367 -2.612839  0.268636  0.965359 -1.409323   
103356   68614.0 -1.106949  0.189437  2.654398 -0.356116 -1.301974  0.315720   
86237    61148.0  0.214494  0.914799  1.626062  3.021434  0.167768  0.569122   
179378  124049.0 -1.181376 -0.215929  0.569703  0.226399  2.228961 -0.055988   
91158    63335.0  1.377562 -1.014995 -0.129834 -1.951419 -0.472644  0.868123   
...          ...       ...       ...       ...       ...       ...       ...   
45449    42352.0 -0.513649 -1.151130  2.959921 -0.739871 -1.152479  2.809651   
35560    38170.0 -2.161250 -1.582002  1.974286  1.073222 -2.913766  1.233037   
109577   71442.0  1.192095  0.638366 -0.552152  0.696413  0.456463 -0.635509   
17102    28439.0 -1.070980  0.706944  1.004662  0.181041 -0.365535  0.209705   
261594  160091.0 -2.228140  2.454174 -2.843007 -1.973992  2.344578  3.107977   

              V7        V8        V9  ...       V21       V22       V23  \
225550  0.968401 -0.557379  0.163413  ...  0.269930  0.945118 -0.188518   
103356 -0.727371  0.853515  1.109656  ...  0.112089  0.613764 -0.010987   
86237   0.459820 -0.021692 -0.629150  ... -0.115453  0.057778  0.245431   
179378 -0.194383  1.332098  0.289250  ...  0.311013  0.981020 -0.228853   
91158  -1.005950  0.277599  0.532098  ... -0.258619 -0.045332 -0.299731   
...          ...       ...       ...  ...       ...       ...       ...   
45449  -1.418777  0.322926 -0.560906  ... -0.137551  1.040085 -0.016417   
35560   2.546364  0.102097 -0.257232  ...  0.834925  0.959667  1.635092   
109577  0.305622 -0.095702 -0.599373  ... -0.061932 -0.067988 -0.080009   
17102   0.688427 -0.250309  0.960850  ... -0.040701  0.701028 -0.015153   
261594 -0.034037  1.264096  1.324320  ...  0.216480  1.020140  0.080164   

             V24       V25       V26       V27       V28  Amount  Class  
225550  0.723746  0.842321  0.096020 -0.085402 -0.081324    0.76      0  
103356  0.734380 -0.420372  0.910815 -0.006458  0.051323   18.55      0  
86237   0.056157 -1.547727 -0.165904  0.144598  0.036902   11.41      0  
179378  0.749028  1.114429  0.049134  0.336089  0.143695  145.00      0  
91158  -1.706067  0.749496 -0.376679  0.109129  0.002361   20.20      0  
...          ...       ...       ...       ...       ...     ...    ...  
45449  -0.894174 -0.714244  0.200378 -0.343510 -0.332290   31.00      0  
35560   0.503032  0.436232 -0.247213 -0.154104  0.151372  801.35      0  
109577 -0.063400  0.505903  0.377564 -0.010727  0.031658    0.76      0  
17102  -0.389726 -0.449938  0.596969  0.282590  0.189817   96.00      0  
261594  0.701881 -0.343123 -0.187199  0.958333  0.741906    0.89      0  

[492 rows x 31 columns]

In [46]:
O = dff[dff["Class"]==1]

In [47]:
O

Time        V1        V2        V3        V4        V5        V6  \
541        406.0 -2.312227  1.951992 -1.609851  3.997906 -0.522188 -1.426545   
623        472.0 -3.043541 -3.157307  1.088463  2.288644  1.359805 -1.064823   
4920      4462.0 -2.303350  1.759247 -0.359745  2.330243 -0.821628 -0.075788   
6108      6986.0 -4.397974  1.358367 -2.592844  2.679787 -1.128131 -1.706536   
6329      7519.0  1.234235  3.019740 -4.304597 -0.047478  3.624201 -1.357746   
...          ...       ...       ...       ...       ...       ...       ...   
279863  169142.0 -1.927883  1.125653 -4.518331  1.749293 -1.566487 -2.010494   
280143  169347.0  1.378559  1.289381  0.052136  1.411850  0.442581 -1.326536   
280149  169351.0 -0.676143  1.126366 -2.213700  0.468308 -1.120541 -0.003346   
281144  169966.0 -3.113832  0.585864  0.052136  1.817092 -0.840618 -2.943548   
281674  170348.0  1.991976  0.158476 -2.583441  0.408670  1.151147 -0.096695   

              V7        V8        V9  ...       V21       V22       V23  \
541    -2.537387  1.391657 -2.770089  ...  0.517232 -0.035049 -0.465211   
623     0.325574 -0.067794 -0.270953  ...  0.661696  0.435477  1.375966   
4920    0.562320 -0.399147 -0.238253  ... -0.294166 -0.932391  0.172726   
6108   -3.496197 -0.248778 -0.247768  ...  0.573574  0.176968 -0.436207   
6329    1.713445 -0.496358 -1.282858  ... -0.379068 -0.704181 -0.656805   
...          ...       ...       ...  ...       ...       ...       ...   
279863 -0.882850  0.697211 -2.064945  ...  0.778584 -0.319189  0.639419   
280143 -1.413170  0.248525 -1.127396  ...  0.370612  0.028234 -0.145640   
280149 -2.234739  1.210158 -0.652250  ...  0.751826  0.834108  0.190944   
281144 -2.208002  1.058733 -1.632333  ...  0.583276 -0.269209 -0.456108   
281674  0.223050 -0.068384  0.577829  ... -0.164350 -0.295135 -0.072173   

             V24       V25       V26       V27       V28  Amount  Class  
541     0.320198  0.044519  0.177840  0.261145 -0.143276    0.00      1  
623    -0.293803  0.279798 -0.145362 -0.252773  0.035764  529.00      1  
4920   -0.087330 -0.156114 -0.542628  0.039566 -0.153029  239.93      1  
6108   -0.053502  0.252405 -0.657488 -0.827136  0.849573   59.00      1  
6329   -1.632653  1.488901  0.566797 -0.010016  0.146793    1.00      1  
...          ...       ...       ...       ...       ...     ...    ...  
279863 -0.294885  0.537503  0.788395  0.292680  0.147968  390.00      1  
280143 -0.081049  0.521875  0.739467  0.389152  0.186637    0.76      1  
280149  0.032070 -0.739695  0.471111  0.385107  0.194361   77.89      1  
281144 -0.183659 -0.328168  0.606116  0.884876 -0.253700  245.00      1  
281674 -0.450261  0.313267 -0.289617  0.002988 -0.015309   42.53      1  

[492 rows x 31 columns]

In [48]:
fd = pd.concat([Z,O])

In [49]:
vdata = fd.sample(984)

In [50]:
vdata

Time        V1        V2        V3        V4        V5        V6  \
229100  145797.0  0.376521 -0.694268  0.701953 -2.144883 -0.997005 -0.712569   
199896  133184.0 -1.212682 -2.484824  0.052136  3.670562 -0.863375 -1.855855   
226814  144808.0 -2.405207  2.943823  0.052136  3.533374  0.007328 -0.112632   
149357   91075.0 -1.855061  1.554964 -1.405809  0.669327 -0.280230  1.178652   
240598  150671.0  1.855442 -0.505858 -1.027760  0.432050 -0.190042 -0.352374   
...          ...       ...       ...       ...       ...       ...       ...   
175162  122221.0  1.596608 -1.080958 -0.149754  0.344366 -1.168466 -0.425209   
157585  110087.0  1.934946  0.650678 -0.286957  3.987828  0.316052 -0.099449   
250761  155054.0 -0.512349  4.827060  0.052136 -0.047478  0.367704 -2.055129   
8842     12093.0 -4.696795  2.693867 -4.475133 -0.047478 -1.556758 -1.549420   
222133  142840.0 -3.613850 -0.922136  0.052136  3.373001 -0.545207 -1.171301   

              V7        V8        V9  ...       V21       V22       V23  \
229100 -0.331175 -0.419622 -1.340764  ... -0.082136  0.384815  0.152080   
199896  1.017732 -0.544704 -1.703378  ...  1.396872  0.092073 -1.492882   
226814 -1.329372  1.709417 -2.322716  ...  0.652683  0.414132  0.023869   
149357 -3.459979 -2.815155  1.242229  ... -0.095308  0.946629 -0.297403   
240598 -0.124625 -0.069953  0.552212  ...  0.034576  0.031416  0.078561   
...          ...       ...       ...  ...       ...       ...       ...   
175162 -0.533580 -0.062259  1.467548  ...  0.164671  0.267439  0.124446   
157585 -0.021483 -0.172327  0.508730  ... -0.173602 -0.190974  0.219976   
250761 -2.935856  1.431008 -0.028004  ...  0.535620 -0.459496 -0.009364   
8842    0.012668  0.553934 -1.498468  ...  0.573898 -0.080163  0.318408   
222133  0.012668  1.517016 -1.775833  ...  0.786787  0.893065  1.034907   

             V24       V25       V26       V27       V28      Amount  Class  
229100 -0.060283 -0.934066 -0.276442 -0.275071 -0.133133   17.200000      0  
199896 -0.204227  0.532511 -0.293871  0.212663  0.431095   66.955618      1  
226814 -0.260616  0.405316  0.029107  0.519807 -0.469537  667.550000      1  
149357 -0.368344  0.198731 -0.078591  0.366709  0.073767   38.760000      1  
240598 -0.471199 -0.203332  0.227865 -0.060574 -0.053875   89.950000      0  
...          ...       ...       ...       ...       ...         ...    ...  
175162  0.019126 -0.528015 -0.114064  0.001312  0.002873  200.000000      0  
157585 -0.216597 -0.136692 -0.129954 -0.050077 -0.051082    1.000000      1  
250761 -1.140436 -0.006445  0.527970  0.558881  0.126517    0.770000      1  
8842   -0.245862  0.338238  0.032271  0.010900  0.608075    0.000000      1  
222133  0.097671 -1.345551 -0.788329  1.055442  0.099971  144.800000      1  

[984 rows x 31 columns]

In [51]:
XX = vdata.drop("Class", axis = 1)

In [52]:
XX

Time        V1        V2        V3        V4        V5        V6  \
229100  145797.0  0.376521 -0.694268  0.701953 -2.144883 -0.997005 -0.712569   
199896  133184.0 -1.212682 -2.484824  0.052136  3.670562 -0.863375 -1.855855   
226814  144808.0 -2.405207  2.943823  0.052136  3.533374  0.007328 -0.112632   
149357   91075.0 -1.855061  1.554964 -1.405809  0.669327 -0.280230  1.178652   
240598  150671.0  1.855442 -0.505858 -1.027760  0.432050 -0.190042 -0.352374   
...          ...       ...       ...       ...       ...       ...       ...   
175162  122221.0  1.596608 -1.080958 -0.149754  0.344366 -1.168466 -0.425209   
157585  110087.0  1.934946  0.650678 -0.286957  3.987828  0.316052 -0.099449   
250761  155054.0 -0.512349  4.827060  0.052136 -0.047478  0.367704 -2.055129   
8842     12093.0 -4.696795  2.693867 -4.475133 -0.047478 -1.556758 -1.549420   
222133  142840.0 -3.613850 -0.922136  0.052136  3.373001 -0.545207 -1.171301   

              V7        V8        V9  ...       V20       V21       V22  \
229100 -0.331175 -0.419622 -1.340764  ... -0.393710 -0.082136  0.384815   
199896  1.017732 -0.544704 -1.703378  ... -0.014857  1.396872  0.092073   
226814 -1.329372  1.709417 -2.322716  ... -0.338707  0.652683  0.414132   
149357 -3.459979 -2.815155  1.242229  ...  0.536916 -0.095308  0.946629   
240598 -0.124625 -0.069953  0.552212  ... -0.031489  0.034576  0.031416   
...          ...       ...       ...  ...       ...       ...       ...   
175162 -0.533580 -0.062259  1.467548  ...  0.229073  0.164671  0.267439   
157585 -0.021483 -0.172327  0.508730  ... -0.344452 -0.173602 -0.190974   
250761 -2.935856  1.431008 -0.028004  ...  0.804101  0.535620 -0.459496   
8842    0.012668  0.553934 -1.498468  ... -0.158971  0.573898 -0.080163   
222133  0.012668  1.517016 -1.775833  ... -0.320541  0.786787  0.893065   

             V23       V24       V25       V26       V27       V28      Amount  
229100  0.152080 -0.060283 -0.934066 -0.276442 -0.275071 -0.133133   17.200000  
199896 -1.492882 -0.204227  0.532511 -0.293871  0.212663  0.431095   66.955618  
226814  0.023869 -0.260616  0.405316  0.029107  0.519807 -0.469537  667.550000  
149357 -0.297403 -0.368344  0.198731 -0.078591  0.366709  0.073767   38.760000  
240598  0.078561 -0.471199 -0.203332  0.227865 -0.060574 -0.053875   89.950000  
...          ...       ...       ...       ...       ...       ...         ...  
175162  0.124446  0.019126 -0.528015 -0.114064  0.001312  0.002873  200.000000  
157585  0.219976 -0.216597 -0.136692 -0.129954 -0.050077 -0.051082    1.000000  
250761 -0.009364 -1.140436 -0.006445  0.527970  0.558881  0.126517    0.770000  
8842    0.318408 -0.245862  0.338238  0.032271  0.010900  0.608075    0.000000  
222133  1.034907  0.097671 -1.345551 -0.788329  1.055442  0.099971  144.800000  

[984 rows x 30 columns]

In [53]:
yy = vdata["Class"]

In [54]:
yy

229100    0
199896    1
226814    1
149357    1
240598    0
         ..
175162    0
157585    1
250761    1
8842      1
222133    1
Name: Class, Length: 984, dtype: int64

In [55]:
from sklearn.model_selection import train_test_split

In [56]:
XXtrain,XXtest,yytrain,yytest = train_test_split(XX,yy,test_size = 0.20)

In [57]:
XXtrain.shape , XXtest.shape , yytrain.shape , yytest.shape

((787, 30), (197, 30), (787,), (197,))

In [58]:
from sklearn.ensemble import AdaBoostClassifier

In [59]:
adb = AdaBoostClassifier()

In [60]:
adb.fit(XXtrain,yytrain)

AdaBoostClassifier()

In [61]:
adb.score(XXtest,yytest)

0.9441624365482234

In [62]:
adb.score(XXtrain,yytrain)

0.9822109275730623

Here i got good score and also our model is not over fitted but what dataset i took is very small .

In [63]:
from sklearn.model_selection import KFold,cross_val_score

In [64]:
kf = KFold()

In [65]:
from sklearn.ensemble import AdaBoostClassifier,BaggingClassifier,GradientBoostingClassifier,RandomForestClassifier

In [66]:
from sklearn.tree import DecisionTreeClassifier

In [67]:
dtc = DecisionTreeClassifier()

In [68]:
abc = AdaBoostClassifier()

In [69]:
bc = BaggingClassifier()

In [70]:
gbc = GradientBoostingClassifier()

In [71]:
rfc = RandomForestClassifier()

In [72]:
alg = [dtc,abc,bc,gbc,rfc]

In [73]:
for i in alg:
    score=cross_val_score(i,XX,yy,cv= kf).mean()
    print(i,score)

DecisionTreeClassifier() 0.891277323111986
AdaBoostClassifier() 0.9207552056355537
BaggingClassifier() 0.920760385372423
GradientBoostingClassifier() 0.933953175178701
RandomForestClassifier() 0.9359888117683621


In [74]:
from xgboost import XGBClassifier

In [75]:
xgb = XGBClassifier()

In [76]:
kf = KFold(n_splits = 10)

In [77]:
score = cross_val_score(xgb,XX,yy,cv= kf).mean()

In [78]:
score

0.9329313543599259